### PR TITLE
feat(devkit): allow to customize overwrite mode in generateFiles

### PR DESCRIPTION
**Pull request by qortex**
Adds a new capability to choose how to handle already existing target files when using a generator.

## Current Behavior
If a target file already exists, `generateFiles` overwrites existing files. It is not always the intended behaviour, forcing to have custom code in generators to check for already existing files.

## Expected Behavior
A new (optional) argument to generateFiles allows to choose the behavior when a target file already exists: overwrite (current & still default behavior), keep, throw.

Note: `nx affected` fails on my setup, so could only check by hand and all tests including new tests pass ok on devkit.

### CODE REVIEW

**Comment by qortex** [comment]
@FrozenPandaz @vsavkin @bcabanes 
Hi, I'm not familiar with the PR review process for Nx. Should this topic be discussed somewhere to get the PR under review? I just submitted it, but maybe I'm missing a step in the process.
Thanks for the amazing work on Nx :)
Best,

**Comment by qortex** [comment]
Ping
Is there something to do to make sure someone is aware of this PR ?
@FrozenPandaz @vsavkin @bcabanes

**Review by AgentEnder — APPROVED** [review]

**Review by isaacplmann — APPROVED** [review]

**Comment by qortex** [comment]
Thanks!
Should I resolve the conflicts, or do maintainers prefer to do it?

**Comment by qortex** [comment]
@AgentEnder @isaacplmann 
Happy to move this forward, how should I proceed?

**Comment by isaacplmann** [comment]
Go ahead and merge the conflicts

**Comment by isaacplmann** [comment]
I agree, they look unrelated.  I relaunched the CI pipeline

**Comment by qortex** [comment]
How should we move forward @isaacplmann ? Now just linux failed, macos passed 🤔

**Comment by AgentEnder** [comment]
I am unable to rebase + push this up since the fork is on an org account rather than a personal user account. Can you rebase with latest? I think the e2e errors should go away.

**Comment by qortex** [comment]
Ah bummer

Just did rebase + push 🤞

**Comment by qortex** [comment]
Looks like it's related to storybook e2e. Should not block this PR right? Or should I do something?
@AgentEnder

**Comment by isaacplmann** [comment]
No, not your fault.  We're working on fixing that e2e test.

**Comment by qortex** [comment]
Ok, just rebased and pushed, looks like tests are green now!

**Review by isaacplmann — APPROVED** [review]

**Comment by qortex** [comment]
@FrozenPandaz @vsavkin  should we merge that before `main` moves ahead again?
That would be great 😄

**Comment by qortex** [comment]
Test seems to be broken again.
How to get this PR merged?
We actually need this facility in our workflow. I respect if you don't plan to merge, just please let me know so that I take appropriate action and implement that in our project instead of the Nx repo.

Thanks and good luck for the new releases!

**Comment by qortex** [comment]
@isaacplmann 
What's the way forward?

File: packages/devkit/src/generators/generate-files.ts
Comment (FrozenPandaz) [inline packages/devkit/src/generators/generate-files.ts:46]: Please add this flag within an `options` object. This will make future flags not a breaking change.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
Thank you for your contribution! The functionality looks good. Could you please make these implementation changes?

**Comment by mandarini** [comment]
@qortex thank you so much for your contribution! Will you have time to address the suggested changes, or should I go ahead and fix these, instead?

**Pull request closed without merging**
